repo_manager/gh/collaborators.py

```diff
@@ -0,0 +1,230 @@
+    repo_collab_dict = {"Users": {}, "Teams": {}}
+    repo_collab_dict["Users"] = {collaborator.login: collaborator for collaborator in repo_collab_users}
+    repo_collab_dict["Teams"] = {collaborator.slug: collaborator for collaborator in repo_collab_teams}
+    perm_diffs = {"Users": {}, "Teams": {}}
+    for collaborator_type in collaborators_to_check_values_on.keys():
+        for collaborator_name in collaborators_to_check_values_on[collaborator_type]:
+            if collaborator_type == "Users":
+                repo_value = getattr(
+                    repo_collab_dict[collaborator_type][collaborator_name].permissions,
+                    config_collaborator_dict[collaborator_name].permission,
+                    None,
+                )
+            else:
+                repo_value = (
+                    getattr(
+                        repo_collab_dict[collaborator_type][collaborator_name],
+                        "permission",
+                        None,
+                    )
+                    == config_collaborator_dict[collaborator_name].permission
+                )
+            if repo_value is not True:
+                perm_diffs[collaborator_type][collaborator_name] = diff_option(
+                    config_collaborator_dict[collaborator_name].permission,
+                    True,
+                    repo_value,
+                )
+
+    if len(perm_diffs["Users"]) == 0:
+        perm_diffs.pop("Users")
+
+    if len(perm_diffs["Teams"]) == 0:
+        perm_diffs.pop("Teams")
+
+    if len(perm_diffs) > 0:
+        diff["diff"] = perm_diffs
+
+    if len(diff) > 0:
+        return False, diff
+
+    return True, None
+
+
+def update_collaborators(repo: Repository, collaborators: list[Collaborator], diffs: dict[str, Any]) -> set[str]:
+    """Updates a repo's environments to match the expected settings
+
+    Args:
+        repo (Repository): [description]
+        environments (List[environment]): [description]
+        diffs (Dictionary[string, Any]): List of all the summarized differences by environment name
+
+    Returns:
+        set[str]: [description]
+    """
+    errors = []
+    users_dict = {
+        collaborator.name: collaborator
+        for collaborator in filter(
+            lambda collaborator: collaborator.type == "User" and collaborator.name,
+            collaborators,
+        )
+    }
+    teams_dict = {
+        collaborator.name: collaborator
+        for collaborator in filter(
+            lambda collaborator: collaborator.type == "Team" and collaborator.name,
+            collaborators,
+        )
+    }
+
+    def switch(collaborator: Collaborator, diff_type: str) -> None:
+        if diff_type == "missing":
+            if collaborator.type == "User":
+                repo.add_to_collaborators(collaborator.name, collaborator.permission)
+            elif collaborator.type == "Team":
+                update_team_permissions(repo, collaborator)
```

Comment:
can't use repo.organization.add_to_team(collaborator.name, repo.full_name, collaborator.permission) because the organization URL contains /users/ where it should have /orgs/